lib/wda/utils.js

```diff
@@ -194,51 +194,93 @@ CODE_SIGN_IDENTITY = ${signingId}
  * @return {string} returns xctestrunFilePath for given device
- * @throws if WebDriverAgentRunner_iphoneos${platformVersion}-arm64.xctestrun for real device
- * or WebDriverAgentRunner_iphonesimulator${platformVersion}-x86_64.xctestrun for simulator is not found @bootstrapPath,
+ * @throws if WebDriverAgentRunner_iphoneos${sdkVersion|platformVersion}-arm64.xctestrun for real device
+ * or WebDriverAgentRunner_iphonesimulator${sdkVersion|platformVersion}-x86_64.xctestrun for simulator is not found @bootstrapPath,
  * then it will throw file not found exception
  */
-async function setXctestrunFile (isRealDevice, udid, platformVersion, bootstrapPath, wdaRemotePort) {
-  let xctestrunDeviceFileName = `${udid}_${platformVersion}.xctestrun`;
-  let xctestrunFilePath = path.resolve(bootstrapPath, xctestrunDeviceFileName);
-
-  if (!await fs.exists(xctestrunFilePath)) {
-    let xctestBaseFileName = isRealDevice ? `WebDriverAgentRunner_iphoneos${platformVersion}-arm64.xctestrun` :
-      `WebDriverAgentRunner_iphonesimulator${platformVersion}-x86_64.xctestrun`;
-    let originalXctestrunFile = path.resolve(bootstrapPath, xctestBaseFileName);
-    if (!await fs.exists(originalXctestrunFile)) {
-      log.errorAndThrow(`if you are using useXctestrunFile capability then you need to have ${originalXctestrunFile} file`);
-    }
-    // If this is first time run for given device, then first generate xctestrun file for device.
-    // We need to have a xctestrun file per device because we cant not have same wda port for all devices.
-    await fs.copyFile(originalXctestrunFile, xctestrunFilePath);
-  }
-
-  let xctestRunContent = await plist.parsePlistFile(xctestrunFilePath);
-
-  let updateWDAPort = {
+async function setXctestrunFile (isRealDevice, udid, platformVersion, sdkVersion, bootstrapPath, wdaRemotePort) {
+  const xctestrunFilePath = await getXctestrunFilePath(isRealDevice, udid, platformVersion, sdkVersion, bootstrapPath);
+  const xctestRunContent = await plist.parsePlistFile(xctestrunFilePath);
+  const updateWDAPort = {
     WebDriverAgentRunner: {
       EnvironmentVariables: {
         USE_PORT: wdaRemotePort
       }
     }
   };
 
-  let newXctestRunContent = _.merge(xctestRunContent, updateWDAPort);
+  const newXctestRunContent = _.merge(xctestRunContent, updateWDAPort);
   await plist.updatePlistFile(xctestrunFilePath, newXctestRunContent, true);
 
   return xctestrunFilePath;
 }
 
+/**
+ * Return the path of xctestrun if it exists
+ * @param {boolean} isRealDevice - Equals to true if the current device is a real device
+ * @param {string} udid - The device UDID.
+ * @param {string} platformVersion - The platform version of OS.
+ * @param {string} sdkVersion - The Xcode SDK version of OS.
+ * @param {string} bootstrapPath - The folder path containing xctestrun file.
+ */
+async function getXctestrunFilePath (isRealDevice, udid, platformVersion, sdkVersion, bootstrapPath) {
```

Comment:
I don't really like functions with 3 or more arguments. What if we wrap `isRealDevice, udid, platformVersion ` into a single object, because they all are actually properties of the particular device under test?